js/glowingbear.js

```diff
@@ -145,9 +145,14 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
     $rootScope.$on('activeBufferChanged', function(event, unreadSum) {
         var ab = models.getActiveBuffer();
 
+        // Discard unread lines above 2 screenfuls. We can click through to get more if needs be
+        // This is to keep GB responsive when loading buffers which have seen a lot of traffic. See issue #859
+        var max_unread_lines = $scope.lines_per_screen * 2;
```

Comment:
That makes the whole calculation utterly pointless. If you add 10 here and then simplify the formulas a bit you end up with
```js
var linesToRemove = ab.lines.length - (2 * $scope.lines_per_screen + 10);
```
much simpler :)